src/main/java/com/boomi/flow/services/boomi/mdh/records/GoldenRecordRepository.java

```diff
@@ -165,27 +161,9 @@ public void delete(ApplicationConfiguration configuration, String universeId, Li
     private List<MObject> update(ApplicationConfiguration configuration, List<MObject> objects, String universeId, String operation) {
         var universe = client.findUniverse(configuration.getAtomHostname(), configuration.getAtomUsername(), configuration.getAtomPassword(), universeId);
 
-        // TODO: This isn't correct - it would be great to be able to get the actual ID field name (or make a global standard named one)
-        String idField = universe.getLayout().getIdXPath()
-                .split("/")
-                [2];
-
-        for (var object : objects) {
-            if (Strings.isNullOrEmpty(object.getExternalId())) {
-                // We're creating this object so let's create an ID
-                var id = UUID.randomUUID().toString();
-
-                // Set the ID property, so it can be referenced in a Flow
-                for (var property : object.getProperties()) {
-                    if (property.getDeveloperName().equals(idField)) {
-                        property.setContentValue(id);
-                    }
-                }
-
-                // Set the object's external ID too, which is only used inside Flow itself
-                object.setExternalId(id);
-            }
-        }
+        objects.stream()
+                .filter(object -> Strings.isNullOrEmpty(object.getExternalId()))
+                .forEach(object -> Entities.AddRandomUniqueId(object, universe.getIdField()));
```

Comment:
Same comment about `forEach` here, and using `map` or `pipe` will allow you to combine streams with `objectsBySource`